site/publish.sh

```diff
@@ -24,3 +26,21 @@ ssh ${RSYNC_USER}@${UPDATES_SITE} "/srv/releases/sync-recent-releases.sh /tmp/up
 # delete old sites
 chmod -R a+r www2
 rsync -acvz www2/ --exclude=/updates --delete ${RSYNC_USER}@${UPDATES_SITE}:/var/www/${UPDATES_SITE}
+
+## TODO: cleanup commands above when https://github.com/jenkins-infra/helpdesk/issues/2649 is ready for production
+# Sync Azure File Share content
+azcopy sync ./www2/ "${UPDATES_FILE_SHARE_URL}" --recursive=true --delete-destination=true --exclude-path="updates"
+
+# Debug
+echo "= azcopy sync done."
+
+# Sync CloudFlare R2 buckets content using the updates-jenkins-io profile, excluding 'updates' folder which comes from tool installer generator
+aws s3 sync ./www2/ s3://"${UPDATES_R2_BUCKETS}"/ --profile updates-jenkins-io --no-progress --delete --exclude="updates/*" --endpoint-url "${UPDATES_R2_ENDPOINT}"
+
+# Debug
+echo "= aws sync done."
+
+# /TIME sync, used by mirrorbits to know the last update date to take in account
+date +%s > ./www2/TIME
+aws s3 cp ./www2/TIME s3://"${UPDATES_R2_BUCKETS}"/ --profile updates-jenkins-io --endpoint-url "${UPDATES_R2_ENDPOINT}"
+azcopy cp ./www2/TIME "${UPDATES_FILE_SHARE_URL}" --overwrite=true
```

Comment:
(not needed immediately) to speed up operations, you can send the sync processes in background, and then wait for their completion. The only downside is that it won't catch their exit code with this method:


```suggestion
# /TIME sync, used by mirrorbits to know the last update date to take in account
date +%s > ./www2/TIME
aws s3 cp ./www2/TIME s3://"${UPDATES_R2_BUCKETS}"/ --profile updates-jenkins-io --endpoint-url "${UPDATES_R2_ENDPOINT}" &
azcopy cp ./www2/TIME "${UPDATES_FILE_SHARE_URL}" --overwrite=true &

# Wait for completion of all background processes spawned by the current script
wait
```

There are other method to get the exit code such as writing it as part of a subshell or using GNU `parallel`.

Reply:
Will likely result in unreadable build logs?

Reply:
> Will likely result in unreadable build logs? Probably needs prefixing of log messages.

Good point, I tend to write the logs of subshell into temp log files and then, after the wait, printing the file contents